lib/sidekiq_autoscale/sidekiq_interface.rb

```diff
@@ -39,7 +44,16 @@ def youngest_worker
     private
 
     def process_set
-      @process_set ||= ::Sidekiq::ProcessSet.new
+      @process_set ||= begin
+        set = ::Sidekiq::ProcessSet.new
+        pool = ENV["SIDEKIQ_POOL"]
+
+        if pool
+          set = set.map {|ps| ps if ps.tag == pool }.compact
+        end
```

Comment:
```suggestion
        set = set.map { |ps| ps if ps.tag == pool }.compact if pool
```
Since this is a short line, you can use the "[expression modifier](https://ruby-doc.org/core-2.1.2/doc/syntax/control_expressions_rdoc.html#label-Modifier+if+and+unless)" if-statement (`var = foo if bar?`) if you want to. This is a matter of taste though, so dealer's choice.

Reply:
```suggestion
        set = set.select { |ps| ps.tag == pool } if pool
```

Also, I forgot to mention this when we chatter yesterday, but since this map block is selectively returning `self`, it's better to use `select`/`filter`.